catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/CatalogServiceImpl.java

```diff
@@ -601,16 +620,37 @@ private CompletionStage<MultiTableUpdate> applyIcebergViewCommitOperation(
     return commitBuilderStage;
   }
 
-  private void validateIcebergOperation(IcebergCatalogOperation icebergOp) {
-    for (IcebergMetadataUpdate u : icebergOp.updates()) {
-      if (u instanceof SetLocation) {
-        StorageUri uri = StorageUri.of(((SetLocation) u).location());
-        checkArgument(objectIO.isValidUri(uri), "Unsupported table or view location: %s", uri);
-        WarehouseConfig warehouse = catalogConfig.getWarehouse(icebergOp.warehouse());
-        boolean inWarehouse = !StorageUri.of(warehouse.location()).relativize(uri).isAbsolute();
-        checkArgument(inWarehouse, "Invalid table or view location : %s", uri);
+  private List<IcebergMetadataUpdate> pruneUpdates(IcebergCatalogOperation op, boolean update) {
+    if (update) {
+      return op.updates();
+    }
+    List<IcebergMetadataUpdate> prunedUpdates = new ArrayList<>(op.updates());
+    String location = null;
+    if (op.hasUpdate(SetProperties.class)) {
+      Map<String, String> properties =
+          op.getSingleUpdateValue(SetProperties.class, SetProperties::updates);
+      if (properties.containsKey(IcebergTableMetadata.STAGED_PROPERTY)) {
+        String stagedLocation = op.getSingleUpdateValue(SetLocation.class, SetLocation::location);
```

Comment:
Let's just use the location from `SetLocation` for now. We can add actual verifications later.